test/videojs-contrib-hls.test.js

```diff
@@ -1096,6 +1096,64 @@ QUnit.test('playlist 404 should blacklist media', function(assert) {
   assert.equal(this.player.tech_.hls.stats.bandwidth, 1e10, 'bandwidth set above');
 });
 
+QUnit.test('blacklists playlist if it has stopped being updated', function(assert) {
+  let media;
+  let url;
+
+  this.player.src({
+    src: 'master.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+  openMediaSource(this.player, this.clock);
+  this.player.tech_.triggerReady();
+
+  this.requests[0].respond(200, null,
```

Comment:
We might be able to use `standardXHRResponse` to save some lines. We can also use `this.requests.shift` to make it so that the latest outstanding request is always at the beginning of the array (no need to keep track of them via indexing).